tecken/api/views.py

```diff
@@ -589,34 +580,28 @@ def upload_files(request):
             include_bucket_names.append(bucket_name)
     if include_bucket_names:
         qs = qs.filter(bucket_name__in=include_bucket_names)
+    return qs
+
+
+def _upload_files_content(request, qs):
+    pagination_form = PaginationForm(request.GET)
+    if not pagination_form.is_valid():
+        raise BadRequest(pagination_form.errors)
```

Comment:
Using `BadRequest` changes the response. With this change, my local dev environment returns this:

```
HTTP 400
Content-Type: application/json

{
  "errors": "<ul class=\"errorlist\"><li>page<ul class=\"errorlist\"><li>Not a number &#x27;abc&#x27;</li></ul></li></ul>"
}
```

Whereas the equivalent request on stage is this:

```
HTTP 400
Content-Type: application/json

{
  "errors": {
    "page": [
      "Not a number 'abc'"
    ]
  }
}
```

I think you want:

```
        return http.JsonResponse({"errors": pagination_form.errors}, status=400)
```

Reply:
I'll fix it and make sure to add a test for it too.
Main reason for raising BadRequest there is that if I return a JsonResponse it's not json serializable (thanks unit tests that caught that!). I can return `{"errors": pagination_form.errors}` and check for `"errors"` in the returned object outside this function. 
How's that sound? lmk if you got another idea

Reply:
fixed